R2 release notes (#84)
nephio-project/docs: content/en/docs/release-notes/R2.md

---
title: Nephio R2 Release Notes
weight: 1
---

## Overview

## Prerequisites

> Please refer to the [Install Guide](https://github.com/nephio-project/docs/tree/main/content/en/docs/guides/install-guides)

Reviewer: I think that it's better to refer to the [BYOC installation document](https://github.com/nephio-project/docs/blob/main/content/en/docs/guides/install-guides/install-on-byoc.md)

Author: Done

> Please refer to the [Install Guide](https://github.com/nephio-project/docs/tree/main/content/en/docs/guides/install-guides)

Reviewer 2: Please use `ref` style references instead of referring to the folders of files in GitHub directly.
I've updated this to use the `ref` style reference and to refer to the prerequisites of the BYOC install guide. 

```suggestion
Please refer to the [Install Guide]({{< ref "docs/guides/install-guides/install-on-byoc.md#prerequisites" >}}) for the
prerequisites on supported environments.
```

Author: Done. Thanks.

for the prerequisites on supported environments.

## Support Matrix

The sandbox environment requires a physical or virtual machine with:

> - Ubuntu-20.04-focal Linux

Reviewer: That restriction is for free5Gc workloads, OAI has been tested on Ubuntu Jammy

Author: thanks @electrocucaracha . So users need different environments for Free5gc and OAI?

Reviewer: Maybe they can use the same, but we have tested with different distros

Author: Can you please mention the exact versions here?

Reviewer 3: @gvbalaji Where are we mentioning about Fedora? Because R1 was working in fedora right?

Author: Added, thanks.

- 8 cores
- 32 GB memory
- 200 GB disk size
- default user with sudo passwordless permissions

This install has been verified on VMs running on Google Cloud.

For non-sandbox installations, any conforming Kubernetes cluster is sufficient
for the Management cluster.

## Features

### API

Common CRDs provided for 5G Core and RAN services

### Web UI

Basic web UI to view and manage the packages and resources within them.

### Packages

* Kpt packages for all [free5gc](https://free5gc.org/) services
* Kpt packages for all [OAI](https://openairinterface.org/) services
* Packages for the core Nephio services
* Packages for the Cluster API services for cluster creation
* Packages for the dependent services

### Functionalities

* Multi-vendor suppor
  * API changes 
  * Generalization 
  * Parameterization
  * Reuse of params 
* Multi-cloud support
  * Sandbox
  * GCP
  * OpenShift
* OAI components

>   * Deploying many OAI 5G core and RAN components (CU and DU) NFs via Python-based operators

Reviewer 3: - Deploying OAI 5G Core and RAN Network Functions (CU-CP, CU-UP, RF Simulated DU)
- OAI 5G Core network functions are developed using Python KOPF (Kubernetes Operator Framework)
- End to End call with RF Simulated OAI-NR-UE

Author: Thanks Done.

  * End-to-end call
* Integration with new release of Porch
  * Stability
  * Security
  * Repo watcher changes
* Documentation

>  new web based documentation 

Reviewer: ```suggestion
  * new web based documentation 
```

Author: Done. Thanks.

---
title: Nephio R2 Release Notes
weight: 1
---

## Overview

## Prerequisites

> Please refer to the [Install Guide]({{< ref "docs/guides/install-guides/install-on-byoc.md#prerequisites" >}})

Reviewer 4: This is a Hugo ref, it won't show in github.com.

for the prerequisites on supported environments.

## Support Matrix

> The sandbox environment requires a physical or virtual machine with:

Reviewer 4: This is for end to end RAN use case only.

- Ubuntu-20.04-focal or Ubuntu 22.04 Linux Or Fedora 34
- 16 cores
- 32 GB memory
- 200 GB disk size
- default user with sudo passwordless permissions

This install has been verified on VMs running on Google Cloud.

> For non-sandbox installations, any conforming Kubernetes cluster is sufficient

Reviewer 4: Reword:

Note: A Nephio sandbox can be created on any Kubernetes cluster.

for the Management cluster.

## Features

### API

Common CRDs provided for 5G Core and RAN services

### Web UI

Basic web UI to view and manage the packages and resources within them.

### Packages

* Kpt packages for all [free5gc](https://free5gc.org/) services
* Kpt packages for all [OAI](https://openairinterface.org/) services
* Packages for the core Nephio services
* Packages for the Cluster API services for cluster creation

> * Packages for the dependent services

Reviewer 4: Typo

### Functionalities

> * Multi-vendor suppor

Reviewer 4: support

  * API changes 
  * Generalization 
  * Parameterization
  * Reuse of params 
* Multi-cloud support
  * Sandbox
  * GCP
  * OpenShift
* OAI components
  * Deploying OAI 5G Core and RAN Network Functions (CU-CP, CU-UP, RF Simulated DU)
  * OAI 5G Core network functions are developed using Python KOPF (Kubernetes Operator Framework)
  * End to End call with RF Simulated OAI-NR-UE

> * Integration with new release of Porch

Reviewer 4: Note that Porch is now part of Nephio, owned by Nephio.

  * Stability
  * Security
  * Repo watcher changes
* Documentation
  * New web based documentation 
* Experimental
  * Nephio SDK
  * Flux-based deployment of Helm charts
  * Helm-to-operator converter

>   * Experimental: Topology Controller - Northbound API to express the network topology

Reviewer 4: Remove "experimental"

### Fixes

* Removing rigid dependency on Gitea

> * Fixing the Multus version

Reviewer 4: Upgrading

  via kubectl to debug any deployment issues.
* Web UI features are limited to view/edit of packages and resources in those
  packages and their deployment. Additional features will be added
  in subsequent releases.
* When the capacities of the UPF,SMF and AMF NFs are changed, the free5gc Operator on the
  Workload cluster will instantiate a new POD with correspondingly modified
  resources (CPU, memory etc.) During this process, the pod will restart. This is a
  limitation of free5gc.
* Only Gitea works with automated cluster provisioning to create new
  repositories and join them to Nephio. To use a different Git provider, you
  must manually provision cluster repositories, register them to the Nephio
  management server, and set up Config Sync on the Workload cluster.
* The WebUI does not require authentication in the current demo configuration.
  Testing of the WebUI with authentication configured has not been done at this
  time.
* The WebUI only shows resources in the default namespace.
* While many types of Git authentication are supported, the testing was only
  done with token-based Git authentication in Gitea.

> ## Known Issues and Workarounds

Reviewer 4: Add these:

* During specialization we may have duplicate paramaterRef extensions, leading to failed deployments (wrong configurations are generated). The current workaround is to redeploy the package. The bug is due to NFDeploy reconciliation not being correctly idempotent.

* The free5GC operator may create duplicate entries in the SMF function configuration. This is harmless, but still incorrect.

> * When deploying the sandbox environment on an Ubuntu VM running on OpenStack,

Reviewer 4: Remove this (or find out what it is about). Ask Francesco?

Reviewer: @lapentad I know that you were playing around OpenStack, have you ever faced this issue. The way that I fixed in test-infra was adding [these lines](https://github.com/nephio-project/test-infra/pull/230/commits/d12b233aafe4496f05545d8516d29c7b3b82be4c), but I'm not sure if this is the same issue.

  the deployment may fail. Reinstalling the packages will get around this issue.

> * Occasionally packages may take a long time to be approved by the auto-approval

Reviewer 4: Open an issue (epic?) for this.

  controller. If they seem stuck, restarting Porch and the Nephio controllers
  may help:
  ```bash
  kubectl -n porch-system rollout restart deploy porch-server
  kubectl -n nephio-system rollout restart deploy nephio-controller
  ```

> * Occasional calls to `kpt alpha rpkg copy` may fail with a message like

Reviewer 4: Open an issue. Change to `porchctl` instead of `kpt alpha`.

---
title: Nephio R2 Release Notes
weight: 1
---

## Overview

## Prerequisites

Please refer to the [Install Guide]({{< ref "docs/guides/install-guides/install-on-byoc.md#prerequisites" >}})
for the prerequisites on supported environments.

## Support Matrix

The sandbox environment requires a physical or virtual machine with:
- Ubuntu-20.04-focal or Ubuntu 22.04 Linux Or Fedora 34
- 16 cores (Needed for end-to-end testing with RAN and core )
- 32 GB memory
- 200 GB disk size
- Default user with sudo passwordless permissions

This installation has been verified on VMs running on Google Cloud.

> A Nephio sandbox can be created on any Kubernetes cluster.

Reviewer: I'm not sure if we should reduce the scope of this statement. Our [End-to-End tests](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/bootstrap/defaults/main.yml#L23) have been performed in `v1.27.1`.
```suggestion
A Nephio sandbox can be created on any [Certified Kubernetes](https://www.cncf.io/training/certification/software-conformance/) v1.27+ cluster.
```
